contracts/entry-point/src/execute.rs

```diff
@@ -182,8 +182,21 @@ pub fn execute_swap_and_action(
     let exact_out = match &user_swap {
         Swap::SwapExactAssetIn(_) => false,
         Swap::SwapExactAssetOut(_) => true,
+        Swap::SmartSwapExactAssetIn(_) => false,
     };
 
+    if let Swap::SmartSwapExactAssetIn(smart_swap) = &mut user_swap {
+        let diff = smart_swap.amount().checked_sub(remaining_asset.amount())?;
+
+        // If the total swap in amount is greater than remaining asset,
+        // adjust the routes to match the remaining asset amount
+        if diff > Uint128::zero() {
+            let largest_route_idx = smart_swap.largest_route_index().unwrap_or(0);
+
+            smart_swap.routes[largest_route_idx].offer_asset.sub(diff)?;
```

Comment:
at this point since we unwrap or default to 0, routes may be empty (since I don't think we do validation here yet) so I'd expect this to panic with index out of range. This would result in an error, but maybe we should explicitly error in when finding the largest route index if it returns none instead of defaulting to 0. wdyt?

Reply:
Yeah I'm down for this